### PR TITLE
t/96: Integrated the Image feature with the ContextualBalloon plugin

### LINKED ISSUE #96: Integration with ContextualBalloon plugin
Image plugin should use [ContextualBalloon](https://github.com/ckeditor/ckeditor5-ui/blob/master/src/contextualballoon.js) to display balloon panels like [Link plugin does](https://github.com/ckeditor/ckeditor5-link/blob/master/src/link.js#L29-L30).

### LINKED ISSUE #106: Alt balloon does not update its position on external changes
apr-24-2017 19-02-59

This may be fixed together with: https://github.com/ckeditor/ckeditor5-image/issues/96

A follow-up of https://github.com/ckeditor/ckeditor5-ui/issues/198.

### LINKED ISSUE #5090: Integration with ContextualBalloon plugin
Image plugin should use [ContextualBalloon](https://github.com/ckeditor/ckeditor5-ui/blob/master/src/contextualballoon.js) to display balloon panels like [Link plugin does](https://github.com/ckeditor/ckeditor5-link/blob/master/src/link.js#L29-L30).

### LINKED ISSUE #5096: Alt balloon does not update its position on external changes
apr-24-2017 19-02-59

This may be fixed together with: https://github.com/ckeditor/ckeditor5-image/issues/96

A follow-up of https://github.com/ckeditor/ckeditor5-ui/issues/198.

### PR DESCRIPTION
**Pull request by oleq**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Type: Integrated the Image feature with the `ContextualBalloon` plugin. Closes ckeditor/ckeditor5#5090. Closes ckeditor/ckeditor5#5096.

### CODE REVIEW

**Comment by oleq** [comment]
I implemented `ImageBalloon#clear/remove` methods, which means https://github.com/ckeditor/ckeditor5-ui/pull/273 is no longer necessary.

**Comment by oleq** [comment]
To fix the following issue, `clickOutsideHandler` should listen on `mousedown` instead of `mouseup` because the later occurs way after `#render` upon which the balloon is being repositioned.

A PR in ckeditor5-ui will be created as a follow-up.

kapture 2017-07-14 at 12 57 25

File: src/image/ui/imageballoon.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+import BalloonPanelView from '@ckeditor/ckeditor5-ui/src/panel/balloon/balloonpanelview';
+import { isImageWidget } from '../utils';
+
+/**
+ * A balloon used by various image features to display toolbars, editing forms
+ * etc.
+ *
+ * @extends module:ui/panel/balloon/contextualballoon~ContextualBalloon
+ */
+export default class ImageBalloon extends ContextualBalloon {
```
Comment (oskarwrobel) [inline src/image/ui/imageballoon.js]: I'm little confused. I don't understand why do we need another instance of contextual balloon. I suppose we don't want to show two balloons at the same time, so it should be possible to use CB for showing image balloons. If you don't want to duplicate some code for two very similar image balloons you can move common code to some helpers.

Comment (oleq) [reply src/image/ui/imageballoon.js]: I guess you're right. I'll rewrite it to use an existing instance of CB.

**Pull request merged**